sw/device/tests/flash_ctrl_rma_test.c

```diff
@@ -0,0 +1,213 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
+#include "sw/device/lib/testing/json/command.h"
+#include "sw/device/lib/testing/json/mem.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/pinmux_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_console.h"
+#include "sw/device/lib/testing/test_framework/ottf_test_config.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf_commands.h"
+#include "sw/device/lib/ujson/ujson.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
+
+volatile uint32_t kTestPhase;
+volatile uint32_t kEndTest;
+
+static dif_uart_t uart;
+static dif_pinmux_t pinmux;
+static dif_flash_ctrl_state_t flash;
+static dif_otp_ctrl_t otp_ctrl;
+
+enum {
+  kFlashInfoBankId = 0,
+  kFlashInfoPartitionId = 0,
+  kFlashInfoPageIdCreatorSecret = 1,
+  kFlashInfoPageIdOwnerSecret = 2,
+  kFlashInfoPageIdIsoPart = 3,
+  kTestPhase0 = 0,
+  kTestPhase1 = 1,
+  kTestPhaseInit = 2,
+};
+
+// Hardcoded secret data for flash info partition.
+static const uint32_t kRandomData[3] = {
+    0xbab0bab1,
+    0xdeadbeef,
+    0xcafefeed,
+};
+
+status_t command_processor(ujson_t *uj) {
```

Comment:
could maybe be replaced by `OTTF_WAIT_FOR`, see https://github.com/lowRISC/opentitan/blob/master/sw/device/lib/testing/test_framework/ottf_utils.h#L26

Reply:
updatead